model/riscv_model.sail

```diff
@@ -0,0 +1,25 @@
+/*=======================================================================================*/
+/*  This Sail RISC-V architecture model, comprising all files and                        */
+/*  directories except where otherwise noted is subject the BSD                          */
+/*  two-clause license in the LICENSE file.                                              */
+/*                                                                                       */
+/*  SPDX-License-Identifier: BSD-2-Clause                                                */
+/*=======================================================================================*/
+
+// Chip reset. This only does the minimum resets required by the RISC-V spec.
+function reset() -> unit = {
+  reset_sys();
+  reset_vmem();
+
+  // To allow model extensions (code outside this repo) to perform additional reset.
+  ext_reset();
+}
+
+// Initialize model state. This is only called once; not for every chip reset.
+function init_model(src: string) -> unit = {
+  assert(config_is_valid(), (if string_length(src) == 0 then "Default config" else "Config in " ^ src)
```

Comment:
Could also do this but I dunno which is preferable. `string_length` is so verbose!

```suggestion
  assert(config_is_valid(), (if src == "" then "Default config" else "Config in " ^ src)
```